### PR TITLE
Adds some useful stuff to the debug outfit

### PR DESCRIPTION
**Pull request by chloecat0**
## About The Pull Request
Gives the debug outfit an ID with every access and a debug hud. Adds alien tools, a holofan, blueprints, a spellbook and a disk with every mutation to the debug box and adds a bunch of missing materials to the material box. Also gives the admin RCD every useable upgrade and adds more parts to the preloaded RPEDs.
Borrows some ideas from tgstation/tgstation#47923

## Why It's Good For The Game
Makes testing stuff a bit easier.

## Changelog
:cl:
admin: Added some useful items to the debug outfit
/:cl:

### CODE REVIEW

**Comment by BrumalLake** [comment]
don't you get some sort of centcom id

**Comment by chloecat0** [comment]
> don't you get some sort of centcom id

The debug outfit has a commander id which doesn't have id console access so having the spare as well could be useful

**Comment by BrumalLake** [comment]
you can't mold the brass and runed metal so

**Comment by KubeRoot** [comment]
The full debug outfit includes additional stuff, including a chief engineer toolbelt, an omnitool, ~~a true AA centcom ID~~, energy axe, T4 BRPED, a human spawner, a hardsuit-mask-oxygen tank setup.

Adding the missing materials is a good idea (at least copper), but not sure about the rest.

Though, as clarified while I was writing the comment, maybe it'd be a better idea to just replace the ID with a full AA one.

**Comment by chloecat0** [comment]
Alright I decided to change some of the items on the debug outfit rather than adding them to the box

**Comment by Archanial** [comment]
I'd rather have CE tools just because alien ones look weird

**Comment by chizzyyyy** [comment]
How about adding more RPED mats too? Right now you only get 10 of each T4.
Also adding a disk with all genetics traits inside too would be wonderful, could you please do that?
Also more than 50 of each mat would be nicer

**Comment by KubeRoot** [comment]
Having silo link on the admin RCD doesn't make much sense, since it already has INFINITE matter.

> I'd rather have CE tools just because alien ones look weird

I agree, maybe a compromise would be having an alien toolbelt in the backpack or box of debug tools? There's advantages to them for sure, especially with alien multitool revealing wires, but I think experimental tools are generally more convenient, especially since you can also fit RPD and RCD on the toolbelt with them.

> Also adding a disk with all genetics traits inside too would be wonderful, could you please do that?

Not sure how well that would work though, I don't think the console has a search functionality?

File: code/game/objects/items/RCD.dm [hunk 1]
```diff
@@ -603,7 +603,7 @@ RLD
 	max_matter = INFINITY
 	matter = INFINITY
 	delay_mod = 0.1
-	upgrade = 1
+	upgrade = RCD_UPGRADE_FRAMES | RCD_UPGRADE_SIMPLE_CIRCUITS | RCD_UPGRADE_SILO_LINK
```
Comment (Mick1299) [inline code/game/objects/items/RCD.dm]: Why does it need the silo link upgrade when it has infinite matter by default

**Comment by KubeRoot** [comment]
I just remembered, I would also suggest porting the debug glasses from https://github.com/tgstation/tgstation/pull/47923, might want to look at the other changes as well to see how they did it. I think it's sometimes nice to have xray easily accessible when debugging.

**Comment by chloecat0** [comment]
Okay, added all those suggestions

**Comment by KubeRoot** [comment]
I got a new suggestion: how about a holofan in the box of tools?

**Comment by BrumalLake** [comment]
all these niche things
don't want something cause it looks weird
who even uses this outfit
you can't even fit the things back into the box

**Comment by KubeRoot** [comment]
> all these niche things

Most of the things in the outfit aren't *that* niche, they're things you might want while debugging stuff.

> don't want something cause it looks weird

Don't want something to *replace the current toolbelt worn by the outfit*, the "looks weird" part is probably also about it just being hard to recognize, and in my case because experimental tools are more compact.

> who even uses this outfit

I do, and if you're not using it while coding stuff you should start, a few clicks and you go from any mob, aghost included, to a human equipped with a hardsuit, oxygen with internals on, a full suite of materials and tools, full access... Often when testing my code I just go straight to switching myself to the debug outfit first thing after spawning in.

> you can't even fit the things back into the box

That is true, but I think that also applies to stuff *actually accessible in-game*, like boxes from an uplink. That said, you do also have a bag of holding to put stuff in, and you can always drop them where you are when just debugging.

*But also, any reason to not make this box also have ridiculously high storage?*

**Comment by BrumalLake** [comment]
i just teleport to where i need to go cause i don't want to walk so i don't need access 99% of the time, god state if i need spaceproofing, and spawn things in cause most of the time it's niche things that you're not gonna find inside the debug outfit
nothing wrong with adding these toys though
you can stay on your mortal plane

File: code/game/machinery/cloning.dm [hunk 2]
```diff
@@ -139,6 +139,16 @@
...
 
+/obj/item/disk/data/debug
+	name = "Debug genetic data disk"
+	desc = "A disk that contains all existing genetic mutations."
+	max_mutations = 100
+
+/obj/item/disk/data/debug/Initialize()
+	. = ..()
+	for(var/i in GLOB.all_mutations)
+		var/datum/mutation/human/HM = i
```
Comment (Archanial) [inline code/game/machinery/cloning.dm]: do a typed loop instead

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/game/machinery/cloning.dm (hunk 2 above)
Comment (KubeRoot) [reply code/game/machinery/cloning.dm]: I believe the code for this is:

```suggestion
	for(var/datum/mutation/human/HM as() in GLOB.all_mutations)
```

Comment (chloecat0) [reply code/game/machinery/cloning.dm]: Yeah that works

**Review by Archanial — APPROVED** [review]

**Pull request merged**